### PR TITLE
create a standalone grpc extension

### PR DESCRIPTION
**Pull request by groob**
this is my windows experiment.

### CODE REVIEW

File: cmd/grpc.ext/grpc.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+		certPins,
+		rootPool,
+		logger,
+	)
+	remote := service.New(conn, level.Debug(logger))
+
+	extOpts := grpcext.ExtensionOpts{
+		EnrollSecret:    enrollSecret,
+		Logger:          logger,
+		LoggingInterval: 30 * time.Second,
```
Comment (marpaia) [inline cmd/grpc.ext/grpc.go]: ideally this could be configurable.

File: cmd/grpc.ext/grpc.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	if err := server.Run(); err != nil {
+		logger.Fatal("err", err)
+	}
+}
+
+type queryier struct {
+	client *osquery.ExtensionManagerClient
+}
+
+func (q *queryier) Query(query string) ([]map[string]string, error) {
```
Comment (marpaia) [inline cmd/grpc.ext/grpc.go]: Copy paste? This doesn't seem to be used.

Comment (groob) [reply cmd/grpc.ext/grpc.go]: It's used on line 37. It's required to make the RequestEnrollment method add initial enrollment data from the client. 

Launcher has this method on the osquery runner.

File: cmd/grpc.ext/grpc.go (hunk 1 above)
Comment (groob) [reply cmd/grpc.ext/grpc.go]: 👍

**Pull request merged**